exec/cnex/gc.c

```diff
@@ -0,0 +1,73 @@
+#include "gc.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#include "bytecode.h"
+#include "cell.h"
+#include "exec.h"
+#include "framestack.h"
+#include "module.h"
+#include "nstring.h"
+#include "stack.h"
+#include "util.h"
+
+
+
+Cell *heap_allocObject(TExecutor *exec)
+{
+    TAllocationObject *o = malloc(sizeof(TAllocationObject));
+    if (o == NULL) {
+        fatal_error("Failed to allocate heap object after %ul allocations.", exec->allocations);
+    }
+    exec->diagnostics.total_allocations++;
+
+    Cell *p = cell_newCell();
+    o->allocNum = exec->diagnostics.total_allocations;
+    o->object = p;
+    o->next = NULL;
+
+    // Push newaly allocated object on the virtual heap.
+    if (exec->firstObject != NULL) {
+        o->next = exec->firstObject;
+    }
+    // Keep track of the start of the heap list.
+    exec->firstObject = o;
+    return p;
+}
+
+unsigned int heap_getObjectCount(TExecutor *exec)
+{
+    // Count all allocated objects, regardless if they are in use or not.
+    unsigned int ret = 0;
+    TAllocationObject *object = exec->firstObject;
+
+    while (object) {
```

Comment:
In this case, use `object != NULL` for better readability.

Reply:
Fixed in 3 locations.